platform/AppStoreConnect/Sources/AppStoreConnect/Builds.swift

```diff
@@ -0,0 +1,94 @@
+import AppStoreConnect_Swift_SDK
+import Foundation
+
+enum BuildStatus {
+    case notYetUploaded
+    case processing
+    case invalid
+    case failed
+    case valid(Build)
+}
+
+extension AppStoreConnect {
+    
+    /// Waits until a given build is processed and is ready to be used.
+    /// - Parameters:
+    ///   - appID: The id of the app for which to search the build for
+    ///   - version: The version of the build to look for
+    ///   - buildNumber: The build number for the given version to look for
+    /// - Returns: The processed build if found, or throws error if something failed
+    func waitUntilBuildIsProcessed(forAppID appID: String,
+                                   withVersion version: String,
+                                   buildNumber: Int) async throws -> Build {
+        let maxNumberOfRequests = 5
```

Comment:
This is the kind of thing that, in a refactoring, could go to a separate configuration file. I foresee a situation where this number of requests is not enough, and the person running it has to check the code to change it. Alternatively, having it as a command line configurable parameter would be nice, with this as a default value. I have no idea how many requests are reasonable, or allowed, but being able to change them by the user might be nice.

Reply:
Could be an option for future improvements.